benchopt/tests/test_cli.py

```diff
@@ -12,3 +13,8 @@ def test_invalid_benchmark():
 def test_invalid_dataset():
     with pytest.raises(SystemExit, match=r"2"):
         run(['lasso', '-d', 'invalid_dataset', '-s', 'baseline'], 'benchopt')
+
+
+def test_invalid_solver():
+    np.testing.assert_raises(ValueError, subprocess.Popen,
+                             'benchopt run lasso -s sdjglsdglxcnv'.split())
```

Comment:
With `pytest`, you should use `pytest.raises` to check for raised Errors.

For the test, you can check the output of Popen like this

```python
p = subprocess.Popen('benchopt run lasso -s sdjglsdglxcnv'.split(),
					 stderr=subprocess.PIPE)
stdout, stderr = p.communicate()
assert 'ValueError' in stderr.decode()
```